validation.go

```diff
@@ -842,9 +835,14 @@ func (v *subSchema) validateNumber(currentSubSchema *subSchema, value interface{
 		}
 	}
 
-	// format
+	v.validateFormat(currentSubSchema, number, result, context)
+
+	result.incrementScore()
+}
+
+func (v *subSchema) validateFormat(currentSubSchema *subSchema, value interface{}, result *Result, context *JsonContext) {
```

Comment:
All the `validate*` functions are designed to validate a specific type of object (with the exception of `validateCommon` which is applied regardless of type). `validateFormat` would break with this philosophy as it's applying a format checking rather than validating a specific type of object. The appropriate place for this code would be in `validateCommon` as the format check has to be applied regardless of the type of object it is that is being validated,